### PR TITLE
Fix for direct memif logic

### PR DESCRIPTION
**Pull request by Mixaster995**
Signed-off-by: Mikhail Avramenko <avramenkomihail15@gmail.com>

## Description
Refactored memif chain element and made separate element which supports direct memif logic

## Issue link
fixes https://github.com/networkservicemesh/sdk-vpp/issues/287

## How Has This Been Tested?
- [ ] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested
## Types of changes
- [x] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

**Comment by Mixaster995** [comment]
@edwarnicke please, share your thoughts on modifications i made for memif

File: pkg/networkservice/mechanisms/memif/server.go [hunk 1]
```diff
@@ -18,31 +18,62 @@ package memif
...
+					_, _ = m.Close(ctx, conn)
+					return nil, err
+				}
+				mechanism.SetSocketFileURL((&url.URL{Scheme: memifMech.SocketFileScheme, Path: memifSocketAddDel.SocketFilename}).String())
+				delete(ctx, true)
+				ifindex.Delete(ctx, true)
+				return conn, nil
+			}
+		}
+	}
```
Comment (denis-tingaikin) [inline pkg/networkservice/mechanisms/memif/server.go:75]: I would be suggesting to not add complexity into memif server element and create a separate chain element for directmemif.

I'm fine with these changes if @edwarnicke has no comments on this.

Comment (edwarnicke) [reply pkg/networkservice/mechanisms/memif/server.go:75]: I'm also inclined to a new chain element for directmemif ... but if that is not possible, lets at least default the directMemif=false

Comment (Mixaster995) [reply pkg/networkservice/mechanisms/memif/server.go:75]: ok, changed default value

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**